arrow/util.py

```diff
@@ -23,6 +28,45 @@ def is_timestamp(value):
         return False
 
 
+def windows_datetime_from_timestamp(timestamp, tz=None):
+    """Computes datetime from timestamp. Supports negative timestamps on Windows platform."""
+    sec_frac, sec = math.modf(timestamp)
+    dt = datetime.datetime(1970, 1, 1, tzinfo=dateutil.tz.tzutc()) + datetime.timedelta(
+        seconds=sec, microseconds=sec_frac * 1e6
+    )
+    if tz is not None:
+        if tz == dateutil.tz.tzlocal():
+            # because datetime.astimezone does not work on Windows for tzlocal() and dates before the 1970-01-01
+            # take timestamp from appropriate time of the year, because of daylight saving time changes
+            ts = time.mktime(dt.replace(year=1970).timetuple())
+            dt += datetime.datetime.fromtimestamp(
+                ts
+            ) - datetime.datetime.utcfromtimestamp(ts)
+            dt = dt.replace(tzinfo=dateutil.tz.tzlocal())
+        else:
+            dt = dt.astimezone(tz)
+    return dt
+
+
+def safe_utcfromtimestamp(timestamp, os_name=os.name):
+    """ datetime.utcfromtimestamp alternative which supports negatvie timestamps on Windows platform."""
+    if os_name == "nt" and timestamp < 0:
+        return windows_datetime_from_timestamp(timestamp, dateutil.tz.tzutc())
+    else:
+        return datetime.datetime.utcfromtimestamp(timestamp)
+
+
+def safe_fromtimestamp(timestamp, tz=None, os_name=os.name):
+    """ datetime.fromtimestamp alternative which supports negatvie timestamps on Windows platform."""
+    if os_name == "nt" and timestamp < 0:
+        if tz is None:
+            # because datetime.fromtimestamp default is local time
+            tz = dateutil.tz.tzlocal()
```

Comment:
Do we want to set this? According to [this SO response](https://stackoverflow.com/questions/18812638/get-timezone-used-by-datetime-datetime-fromtimestamp), the tzinfo is None by default in `fromtimestamp`.

Reply:
It is None, but the `fromtimestamp` returns datetime adjusted for the local time, so without it `windows_datetime_from_timestamp` won't return proper datetime. I could remove the tz after the call to `windows_datetime_from_timestamp`, but I don't think it spoils anything or am I wrong?

Reply:
Hmm @systemcatch what do you think?